src/gui/redux/docgen/docgen.api.spec.js

```diff
@@ -0,0 +1,59 @@
+/* eslint-env jest */
+import { fetchDocgen } from './docgen.api'
+
+const dispatch = jest.fn()
+
+describe('gui/redux/docgen', () => {
+  beforeEach(() => {
+    dispatch.mockClear()
+  })
+
+  it('should dispatch RESET_DOCGEN & SET_DOCGEN with response', async () => {
+    // mocks
+    window.fetch = jest.fn(async () => ({
+      json: async () => [{ foo: 'bar' }],
+    }))
+
+    // execute
+    await fetchDocgen()(dispatch)
+
+    // asserts
+    expect(dispatch.mock.calls.length).toBe(2)
```

Comment:
I can't use snapshot on the `fetchDocgen` because it returns nothing (void method). It only execute dispatch method on different cases. Are you suggesting to use snapshot on the dispatch method ?